src/modules/ethereum/transferTransactions.js

```diff
@@ -124,37 +117,46 @@ export async function fetchTransfersTx(account, fromBlock, toBlock) {
             [...logsFrom, ...logsTo].map(async logData => lookupBlockTimestamp(logData, web3))
         );
 
-        return processTransferEvents(logs, account);
+        return processTransferLogs(logs, account);
     } catch (error) {
         throw new Error("fetchTransferList failed.\n" + error);
     }
 }
 
-export function processTransferEvents(eventList, account) {
-    let transfers = eventList.map(event => formatEvent(account, event, event.returnValues));
-    transfers = Array.from(new Set(aggregateSameHashes(transfers)));
-    transfers.sort((a, b) => b.blockNumber - a.blockNumber);
-    return transfers;
+export function processTransferLogs(events, account) {
+    const transfers = events.map(event => formatEvent(event, account));
+
+    return aggregateAndSortTransfers(transfers);
 }
 
 // called from augminTokenProvider, arguments are in ethers event listener format
-export async function processNewTransferTx(account, eventObject) {
+export async function processNewTransferEvent(event, account) {
+    const transfers = store.getState().userTransfers.transfers;
     const web3 = store.getState().web3Connect.web3Instance;
-    eventObject = await lookupBlockTimestamp(eventObject, web3);
-    return formatEvent(account, eventObject, eventObject.args);
+    const eventWithBlockInfo = await lookupBlockTimestamp(event, web3);
+    const newTransfer = formatEvent(eventWithBlockInfo, account);
+
+    return aggregateAndSortTransfers([...transfers, newTransfer]);
+}
+
+function aggregateAndSortTransfers(transfers) {
+    transfers = Array.from(new Set(aggregateSameHashes(transfers)));
+    transfers.sort((a, b) => b.blockNumber - a.blockNumber);
+
+    return transfers;
 }
 
 const DELEGATED_NARRATIVE = "Delegated transfer fee";
 
-// get txData in format of logData returned from web3.getPastEvents or with eventObject passed by ethers event listener
-function formatEvent(account, txData, augmintTransferEvent) {
-    const logData = Object.assign({ args: augmintTransferEvent }, txData, {
-        from: augmintTransferEvent.from.toLowerCase(),
-        to: augmintTransferEvent.to.toLowerCase(),
-        fee: augmintTransferEvent.fee * -1,
-        amount: augmintTransferEvent.amount * 1,
-        narrative: augmintTransferEvent.narrative,
-        key: `${txData.transactionHash}-${txData.transactionIndex}`
+// get txData in format of logData returned from web3.getPastEvents or with event from event listener
+function formatEvent(event, account) {
+    const logData = Object.assign({}, event, {
+        from: event.returnValues.from.toLowerCase(),
```

Comment:
same as above, would be better to retain the original `augmintTransferEvent` references (less diff).

Reply:
not sure here it could work as there is formatting happening here